src/transformState.js

```diff
@@ -5,7 +5,31 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        if (typeof action.extraData === 'object') {
+          Object.assign(state, action.extraData);
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+      case 'clear':
+        for (const key in state) {
+          if (state.hasOwnProperty(key)) {
+            delete state[key];
+          }
+        }
+        break;
+      default:
+        throw new Error(`Error: ${action.type}`);
```

Comment:
I wouldn't throw an error here, because this way you expect that every consumer of this methods handles Errors. 
You could just: 

```suggestion
        console.error(`Error: ${action.type} is incorrect action type`);
```
We typically throw in `async` code, that will be taught later in the course. 

It's not inherently bad to throw, but there is no need for it here.